### PR TITLE
[ads] Allow non-Rewards Android users to clear Brave Ads data from `Delete browsing data`

### PR DESCRIPTION
**Pull request by aseren**
The PR adds `Clear Brave Ads data...` and `Reset Brave Rewards data...` links to the `Delete browsing data` dialog in settings. The Desktop has similar links for `Delete browsing data` settings.

`Clear Brave Ads data...` link is shown for non-Rewards users. Clicking the link deletes Brave Ads data from ads_service directory in Profile and restarts Brave Ads service:

image

`Reset Brave Rewards data...` link is shown for Rewards users. Clicking the link opens a new tab with `brave://rewards/#reset` URL:

image

Resolves https://github.com/brave/brave-browser/issues/40290

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Test case 1
* Fresh install
* Make sure that Brave Ads service is started.
     EXPECTATION: There should be log line:
    `Successfully initialized ads`
* Clear the Rewards log
* Open `Settings` -> `History` -> `Delete browsing data`
* Tap `Clear Brave Ads data...` link
* Make sure that Brave Ads service is restarted
     EXPECTATION: There should be log line:
    `Initializing ads`
* Check Brave Ads database was deleted and recreated:
     EXPECTATION: There should be log line:
    `Created database for schema version `
* Check that `Brave Ads` client state file was deleted and recreated:
     EXPECTATION: There should be log line:
    `Client state does not exist, creating default state`
* Check that `Brave Ads` confirmation state file was deleted and recreated:
     EXPECTATION: There should be log line:
    `Confirmation state does not exist, creating default state`

### Test case 2
* Fresh install
* Join Brave News
* Make sure that Brave Ads service is started.
     EXPECTATION: There should be log line:
    `Successfully initialized ads`
* Clear the Rewards log
* Open `Settings` -> `Brave Shields & privacy` -> `Delete browsing data` -> `Advanced` tab
* Tap `Clear Brave Ads data...` link
* Make sure that Brave Ads service is restarted
     EXPECTATION: There should be log line:
    `Initializing ads`
* Check Brave Ads database was deleted and recreated:
     EXPECTATION: There should be log line:
    `Created database for schema version `
* Check that `Brave Ads` client state file was deleted and recreated:
     EXPECTATION: There should be log line:
    `Client state does not exist, creating default state`
* Check that `Brave Ads` confirmation state file was deleted and recreated:
     EXPECTATION: There should be log line:
    `Confirmation state does not exist, creating default state`

### Test case 3
* Fresh install
* Join Brave Rewards
* Open `Settings` -> `History` -> `Delete browsing data`
* Make sure `Reset Brave Rewards data...` link is shown
* Open `Settings` -> `Brave Shields & privacy` -> `Delete browsing data` -> `Advanced` tab
* Make sure `Reset Brave Rewards data...` link is shown

### CODE REVIEW

File: components/brave_ads/browser/ads_service_impl.cc [hunk 1]
```diff
@@ -1292,6 +1292,10 @@ void AdsServiceImpl::PurgeOrphanedAdEventsForType(
   }
 }
 
+void AdsServiceImpl::ClearData() {
+  LOG(ERROR) << "TO BE IMPLEMENTED";
```
Comment (SergeyZhukovsky) [inline components/brave_ads/browser/ads_service_impl.cc:1296]: just checking. Is it planned to implement it in a follow up or it's missed?

File: build/android/bytecode/java/org/brave/bytecode/BraveClearBrowsingDataFragmentAdapter.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+public class BraveClearBrowsingDataFragmentAdapter extends BraveClassVisitor {
+    static String sFragment =
+            "org/chromium/chrome/browser/browsing_data/ClearBrowsingDataFragmentAdvanced";
+    static String sBraveFragment =
+            "org/chromium/chrome/browser/browsing_data/BraveClearBrowsingDataFragmentAdvanced";
+
+    public BraveClearBrowsingDataFragmentAdapter(ClassVisitor visitor) {
+        super(visitor);
+        redirectConstructor(sFragment, sBraveFragment);
```
Comment (SergeyZhukovsky) [inline build/android/bytecode/java/org/brave/bytecode/BraveClearBrowsingDataFragmentAdapter.java:18]: Please add a test similar to that one https://github.com/brave/brave-core/blob/1.68.x/android/javatests/org/chromium/chrome/browser/BytecodeTest.java#L1314 for constructors match

File: components/brave_ads/browser/ads_service_impl.cc (hunk 1 above)
Comment (aseren) [reply components/brave_ads/browser/ads_service_impl.cc]: This is added in the PR https://github.com/brave/brave-core/pull/25050, which I'm waiting to be merged soon.
When the PR is merged then the current PR will be rebased.

**Review by samartnik — APPROVED** [review]
java patching looks good

File: android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java [hunk 3]
```diff
@@ -111,7 +111,6 @@ public class BraveRewardsPanel
     private static final String SUPPORT_URL = "https://support.brave.com";
     private static final String BRAVE_REWARDS_PAGE = "https://brave.com/rewards";
     private static final String BRAVE_REWARDS_CHANGES_PAGE = "https://brave.com/rewards-changes";
-    private static final String BRAVE_REWARDS_RESET_PAGE = "brave://rewards#reset";
```
Comment (aseren) [inline android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java:114]: Moved `BRAVE_REWARDS_RESET_PAGE` to `BraveActivity` to make it accessible from other code.

File: components/brave_rewards/resources/shared/lib/rewards_urls.ts [hunk 4]
```diff
@@ -6,8 +6,8 @@
 export const termsOfServiceURL = 'https://basicattentiontoken.org/user-terms-of-service'
 export const privacyPolicyURL = 'https://brave.com/privacy/#rewards'
 export const settingsURL = 'chrome://rewards'
-export const connectURL = 'chrome://rewards#verify'
-export const resetURL = 'chrome://rewards#reset'
+export const connectURL = 'chrome://rewards/#verify'
```
Comment (aseren) [inline components/brave_rewards/resources/shared/lib/rewards_urls.ts:9]: Changed URLs here to fit what have in BraveActivity class.

File: build/android/bytecode/java/org/brave/bytecode/BraveClearBrowsingDataFragmentAdapter.java (hunk 2 above)
Comment (aseren) [reply build/android/bytecode/java/org/brave/bytecode/BraveClearBrowsingDataFragmentAdapter.java:18]: Added

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by tmancey — APPROVED** [review]
LGTM++

**Review by deeppandya — APPROVED** [review]
LGTM

**Pull request merged**